### PR TITLE
Add support for rich media templates

### PR DESCRIPTION
**Pull request by Soufyanbreda**

### CODE REVIEW

File: src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateMessage.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+package com.cmtelecom.text.sdk.models.templates;
+
+import com.cmtelecom.text.sdk.models.multichannel.IRichMessage;
+import com.google.gson.annotations.SerializedName;
+
+public class TemplateMessage implements IRichMessage {
+     /// <summary>
+     /// The Content of the WhatsApp template message
+        /// </summary>
```
Comment (EricSmekens) [inline src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateMessage.java]: Could you re-check all formatting?

File: src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateComponents.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import com.cmtelecom.text.sdk.models.templates.TemplateParameters;
+import com.google.gson.annotations.SerializedName;
+
+
+public class TemplateComponents {
+    /// <summary>
+    /// Source: https://developers.facebook.com/docs/whatsapp/api/messages/message-templates
+    /// 
+    /// Default text if localization fails
```
Comment (EricSmekens) [inline src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateComponents.java]: Could you recheck the documentation, you could use the documentation in the js-version.

File: src/main/java/com/cmtelecom/text/sdk/models/templates/LocalizableParam.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 soufyan.bargach.
```
Comment (EricSmekens) [inline src/main/java/com/cmtelecom/text/sdk/models/templates/LocalizableParam.java]: Going to be super nit-picky here.
Can you set the Copyright to CM.com?

Not sure what Java-standards are, but normally a LICENSE file in the root specifies this, so you can leave this all out of each file.

Comment (Soufyanbreda) [reply src/main/java/com/cmtelecom/text/sdk/models/templates/LocalizableParam.java]: ah this is something that's get generated when creating a new class will delete those comments

Comment (Soufyanbreda) [reply src/main/java/com/cmtelecom/text/sdk/models/templates/LocalizableParam.java]: i have removed those comments ⚡️

File: src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateComponents.java (hunk 2 above)
Comment (Soufyanbreda) [reply src/main/java/com/cmtelecom/text/sdk/models/templates/TemplateComponents.java]: have updated the documentation 🥇

**Review by EricSmekens — APPROVED** [review]

**Pull request merged**